web/src/test/java/com/graphhopper/http/resources/RouteResourceTest.java

```diff
@@ -502,6 +504,54 @@ public void testGPXWithError() {
         assertTrue(str.contains("<hints><error details=\"java"), "Expected error but was: " + str);
     }
 
+    @ParameterizedTest
+    @CsvSource({
+            "false, -1",
+            "true,0",
+            "true,1000",
+    })
+    public void testGPXExport(boolean usePost, int maxUnzippedLength) {
+        GHRequest req = new GHRequest(42.554851, 1.536198, 42.510071, 1.548128);
+        req.putHint("elevation", false);
+        req.putHint("instructions", true);
+        req.putHint("calc_points", true);
+        req.putHint("gpx.millis", "300000000");
+        req.putHint("type", "gpx");
+        GraphHopperWeb gh = new GraphHopperWeb(clientUrl(app, "/route"))
+                .setPostRequest(usePost)
+                ._setMaxUnzippedLength(maxUnzippedLength);
```

Comment:
We could also remove maxunzipped length here since we are not testing it anyway, but we might need it for some other tests in `GraphHopperWebIT` (but then we can still add it -> removed it here :))